app/src/main/java/org/sopt/korailtalk/presentation/home/HomeScreen.kt

```diff
@@ -1,55 +1,146 @@
+    GridItemData(R.drawable.ic_bus, "공항버스"),
+    GridItemData(R.drawable.ic_car, "렌터카"),
+    GridItemData(R.drawable.ic_car_sharing, "카셰어링"),
+    GridItemData(R.drawable.ic_carrier, "짐배송"),
+    GridItemData(R.drawable.ic_bakery, "카페&빵"),
+    GridItemData(R.drawable.ic_ticket, "레저이용권"),
+    GridItemData(R.drawable.ic_taxi, "관광택시")
+)
+
 @Composable
-private fun HomeScreen(
-    onReservationClick: () -> Unit,
-    modifier: Modifier = Modifier
+private fun HomeScreenContent(
+    modifier: Modifier = Modifier,
+    viewModel: HomeViewModel = hiltViewModel(),
+    navigateToReservation: (String, String) -> Unit
 ) {
+    val uiState by viewModel.uiState.collectAsState()
+
+
     Column(
-        horizontalAlignment = Alignment.CenterHorizontally,
-        verticalArrangement = Arrangement.Center,
-        modifier = modifier
-            .background(Color.White)
-            .fillMaxSize()
+        modifier
+            .background(KorailTalkTheme.colors.gray50)
+            .fillMaxSize(),
+        horizontalAlignment = Alignment.Start,
+        verticalArrangement = Arrangement.Top,
     ) {
-        Text(
-            "Home",
-            modifier = Modifier.noRippleClickable(
-                onClick = onReservationClick
-            )
+        KorailTalkBasicTopAppBar(
+            navigationIcon = {
+                Icon(
+                    painter = painterResource(id = R.drawable.img_korail_logo),
+                    contentDescription = null,
+                    tint = KorailTalkTheme.colors.white
+                )
+            },
+            actions = {
+                Icon(
+                    painter = painterResource(id = R.drawable.ic_translate),
+                    contentDescription = null,
+                    tint = KorailTalkTheme.colors.white
+                )
+                Icon(
+                    painter = painterResource(id = R.drawable.ic_hamburger),
+                    contentDescription = null,
+                    tint = KorailTalkTheme.colors.white
+                )
+            }
         )
+        Row(
+            Modifier.padding(start = 24.dp, top = 24.dp, end = 24.dp, bottom = 16.dp)
+        ) {
+            Text(
+                "어디로 갈까요,",
+                style = KorailTalkTheme.typography.headline.head4M18,
+                color = KorailTalkTheme.colors.primary700,
+            )
+            Text(
+                "민주 님?",
+                style = KorailTalkTheme.typography.headline.head3Sb18,
+                color = KorailTalkTheme.colors.primary700,
+            )
+        }
+        Box(
+            Modifier
+                .fillMaxWidth()
+                .padding(horizontal = 16.dp, vertical = 20.dp),
+            contentAlignment = Alignment.Center
+        ) {
+            CheckTrainCard(
+                startStation = uiState.startStation,
+                endStation = uiState.endStation,
+                onSwapClick = { viewModel.swapStations() }, // 뷰모델의 함수 호출
```

Comment:
```suggestion
                onSwapClick = viewModel::swapStations
```
식으로도 써줄 수 있습니다ㅎㅎ

차이점은 [꿀팁 공유](https://www.notion.so/2a61e48dd960806192e9e668f0a42351?source=copy_link)에 있습니다